spurious percent-encoding in output

Hi,
I'm using opam 2.2.0~beta3 on windows. I've noticed that a package name is percent-encoded in an error message:

```
C:\Users\etien\src\dune>opam install -y lwt cinaps core_bench "csexp>=1.3.0" js_of_ocaml js_of_ocaml-compiler "mdx>=2.3.0" menhir ocamlfind ocamlformat.0.26.1 "odoc>=2.4.0" "ppx_expect>=v0.16.0" ppx_inline_test
 ppxlib ctypes "utop>=2.6.0" "melange>=4.0.0-414"
[ERROR] Package conflict!
  * No agreement on the version of ppx%5fexpect:
    - ppx_expect >= v0.16.0
    - core_bench → ppx_jane < v0.15 → ppx_expect < v0.15
  * No agreement on the version of ocaml-base-compiler:
    - (invariant) → ocaml-base-compiler = 4.14.1
    - core_bench → core = 109.24.00 → ocaml < 4.01.0 → ocaml-base-compiler = 3.09.1
```

The first occurrence is displayed as `ppx%5fexpect`, while the rest is displayed correctly as `ppx_expect`.
Maybe that's a styling issue, since the invalid one is printed in bold while the other ones are using normal text (this on windows Terminal).

![image](https://github.com/ocaml/opam/assets/496345/959cbe0b-4ee2-40cf-b15f-9bce3e2659d1)

This isn't Windows-specific:

```
opam@eb69dfc8d71b:~$ opam-dev install core_bench.v0.15.0 "ppx_expect>=v0.16.0"
[ERROR] Package conflict!
  * No agreement on the version of ppx%5fexpect:
    - ppx_expect >= v0.16.0
    - core_bench = v0.15.0 -> ppx_jane < v0.16 -> ppx_expect < v0.15
  * Incompatible packages:
```
IIRC the `_` is not valid in CUDF, so it looks as though we're not translating a package name back in the "No agreement on the version" part of the message.

--- a/master_changes.md
+++ b/master_changes.md
@@ -26,6 +26,7 @@ users)
 ## Actions
 
 ## Install
+  * Fix package name display for no agreement conflicts [#6055 @rjbou - fix #6030]
 
 ## Remove
 

--- a/master_changes.md
+++ b/master_changes.md
@@ -109,6 +109,7 @@ users)
   * cli versioning: untie output from current major version [#6045 @rjbou]
   * Set `opam-version` to 2.2 for some conflict message tests based on opam repository to stabilise their output [#6045 @rjbou]
   * [BUG]: head -c is not posix compliant. Use cut -b instead. [#5989 @madroach]
+  * Add bad cudf package name encoding (dose3 lib) [#6055 @rjbou]
 
 ### Engine
 

--- a/src/solver/opamCudf.ml
+++ b/src/solver/opamCudf.ml
@@ -426,7 +426,7 @@ module Package = struct
   type t = Cudf.package
   include Dose_common.CudfAdd
   let to_string = string_of_package
-  let name_to_string t = t.Cudf.package
+  let name_to_string t = OpamPackage.name_to_string (cudf2opam t)
   let version_to_string t = string_of_int t.Cudf.version
   let to_json = to_json
   let of_json = of_json

--- a/tests/reftests/dune.inc
+++ b/tests/reftests/dune.inc
@@ -543,6 +543,24 @@
    (run ./run.exe %{exe:../../src/client/opamMain.exe.exe} %{dep:empty-conflicts-006.test} %{read-lines:testing-env}))))
 
 (rule
+ (alias reftest-empty-conflicts-007)
+ (action
+  (diff empty-conflicts-007.test empty-conflicts-007.out)))
+
+(alias
+ (name reftest)
+ (deps (alias reftest-empty-conflicts-007)))
+
+(rule
+ (targets empty-conflicts-007.out)
+ (deps root-N0REP0)
+ (package opam)
+ (action
+  (with-stdout-to
+   %{targets}
+   (run ./run.exe %{exe:../../src/client/opamMain.exe.exe} %{dep:empty-conflicts-007.test} %{read-lines:testing-env}))))
+
+(rule
  (alias reftest-env)
  (action
   (diff env.test env.out)))

--- a/tests/reftests/empty-conflicts-007.test
+++ b/tests/reftests/empty-conflicts-007.test
@@ -16,7 +16,7 @@ opam-version: "2.0"
 ### opam switch create --empty encode-cudf
 ### opam install bar foo
 [ERROR] Package conflict!
-  * No agreement on the version of ba%5fz:
+  * No agreement on the version of ba_z:
     - bar -> ba_z < 2
     - foo -> ba_z >= 3
 

--- a/tests/reftests/empty-conflicts-007.test
+++ b/tests/reftests/empty-conflicts-007.test
@@ -1,0 +1,24 @@
+N0REP0
+### : Test the output of 'No agreement' package name, du to Dose_Common.package encoding
+### : see https://github.com/ocaml/opam/issues/6030
+### <pkg:foo.1>
+opam-version: "2.0"
+depends: "ba_z" { > "2" }
+### <pkg:bar.1>
+opam-version: "2.0"
+depends: "ba_z" { < "2" }
+### <pkg:ba_z.1>
+opam-version: "2.0"
+### <pkg:ba_z.2>
+opam-version: "2.0"
+### <pkg:ba_z.3>
+opam-version: "2.0"
+### opam switch create --empty encode-cudf
+### opam install bar foo
+[ERROR] Package conflict!
+  * No agreement on the version of ba%5fz:
+    - bar -> ba_z < 2
+    - foo -> ba_z >= 3
+
+No solution found, exiting
+# Return code 20 #

